terraform/environments/corporate-staff-rostering/ssm-documents/ami-build-command.yaml

```
@@ -0,0 +1,75 @@
+---
+schemaVersion: "2.2"
+description: "Powershell doc for MGN-replicated Azure VMs." 
+mainSteps:
+  - name: installAWSToolsForPowershell
+    action: aws:runPowerShellScript
+    inputs:
+      runCommand:
+        - |
+          [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+          Import-Module AWSPowerShell -Force
+
+          $module = Get-Module -ListAvailable -Name AWSPowerShell
+          if (-not $module) {
+            Write-Host "AWS Powershell Toolkit is not installed, installing now..."
+            Import-Module AWSPowerShell -Force
+          } else {
+            Write-Host "AWS Powershell Toolkit is already installed." 
+          }
+
+  - name: activeDirectoryModulePresent
+    action: aws:runPowerShellScript
+    inputs: 
+      runCommand:
+        - |
+          $module = Get-Module -ListAvailable -Name ActiveDirectory
+          if (-not $module) {
+            Write-Host "Active Directory module is not installed, installing now..."
+            Install-WindowsFeature -Name "RSAT-AD-PowerShell" -IncludeAllSubFeature
+          } else {
+            Write-Host "Active Directory module is already installed." 
+          }
+
+  - name: getRootVolumeSpace
+    action: aws:runPowerShellScript
+    maxAttempts: 1
+    inputs:
+      runCommand:
+        - |
+          $rootVolume = Get-WmiObject -Query "SELECT * FROM Win32_LogicalDisk WHERE DeviceID = 'C:'"
+          $availableSpaceGB = [math]::Round(($rootVolume.FreeSpace / 1GB), 4)
+          $totalSpaceGB = [math]::Round(($rootVolume.Size / 1GB), 4) # good 
+          $availablePercentage = [math]::Round((($availableSpaceGB / $totalSpaceGB) * 100), 4)
+
+          Write-Output "Available space on the C drive is $availableSpaceGB GB, which is $availablePercentage% % of the total space"
+          
+          if ($availablePercentage -lt 25) {
+            Write-Output "Available disk space is less than 25 %; stopping AMI build."
+            Exit 1
+          } else {
+            Write-Output "Available disk space is more than 25 %; Continuing with AMI build"
+          }
+
+  - name: awsDiscoveryAgent
+    action: aws:runPowerShellScript 
+    inputs:
+      runCommand:
+        - |
+          $output = Try {
+          Stop-Service -Name 'AWSDiscoveryService'-ErrorAction Stop 
+          } Catch {
```

Indent for readability.
```suggestion
          $output = Try {
            Stop-Service -Name 'AWSDiscoveryService'-ErrorAction Stop 
          } Catch {
```